### PR TITLE
Fix a few clang warnings

### PR DESCRIPTION
**Pull request by bernhardmgruber**
This PR fixes and suppresses warnings reported by clang, ordinarily, and by clang on nvcc-generated code.

### CODE REVIEW

File: test/common/devCompileOptions.cmake [hunk 1]
```diff
@@ -150,13 +150,22 @@ if(CMAKE_CXX_COMPILER_ID MATCHES "Clang" OR CMAKE_CXX_COMPILER_ID STREQUAL "Inte
...
+    list(APPEND alpaka_DEV_COMPILE_OPTIONS "$<$<COMPILE_LANG_AND_ID:CUDA,NVIDIA>:SHELL:-Xcompiler -Wno-deprecated>")
+    if(CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 13.0)
+        list(APPEND alpaka_DEV_COMPILE_OPTIONS "$<$<COMPILE_LANG_AND_ID:CUDA,NVIDIA>:SHELL:-Xcompiler -Wno-reserved-identifier>")
+    endif()
+    if(CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 15.0)
+        list(APPEND alpaka_DEV_COMPILE_OPTIONS "$<$<COMPILE_LANG_AND_ID:CUDA,NVIDIA>:SHELL:-Xcompiler -Wno-gnu-line-marker>")
+    endif()
+    if(CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 10.0)
+        list(APPEND alpaka_DEV_COMPILE_OPTIONS "$<$<COMPILE_LANG_AND_ID:CUDA,NVIDIA>:SHELL:-Xcompiler -Wno-deprecated-copy>")
+    endif()
```
Comment (j-stephan) [inline test/common/devCompileOptions.cmake:168]:
```suggestion
    if(CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 14.0)
        list(APPEND alpaka_DEV_COMPILE_OPTIONS "$<$<COMPILE_LANG_AND_ID:CUDA,NVIDIA>:SHELL:-Xcompiler -Wno-reserved-identifier>")
        list(APPEND alpaka_DEV_COMPILE_OPTIONS "$<$<COMPILE_LANG_AND_ID:CUDA,NVIDIA>:SHELL:-Xcompiler -Wno-deprecated-copy>")
    endif()
    if(CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 15.0)
        list(APPEND alpaka_DEV_COMPILE_OPTIONS "$<$<COMPILE_LANG_AND_ID:CUDA,NVIDIA>:SHELL:-Xcompiler -Wno-gnu-line-marker>")
    endif()
```

We don't support clang-cuda below clang-14.

**Review by j-stephan — CHANGES_REQUESTED** [review]

File: test/common/devCompileOptions.cmake (hunk 1 above)
Comment (bernhardmgruber) [reply test/common/devCompileOptions.cmake:168]: But nvcc + clang below 14.

Comment (j-stephan) [reply test/common/devCompileOptions.cmake:168]: Ugh, right. My bad.

**Review by j-stephan — APPROVED** [review]

File: test/common/devCompileOptions.cmake (hunk 1 above)
Comment (bernhardmgruber) [reply test/common/devCompileOptions.cmake:168]: Ah, I see the source of confusion. The warning happens with nvcc + clang, so the `CMAKE_CUDA_HOST_COMPILER` is clang. The `CMAKE_CXX_COMPILER` can be any other compiler, but for alpaka is usually the same.

**Pull request merged**